botocore/parsers.py

```diff
@@ -507,18 +507,18 @@ def _handle_timestamp(self, shape, value):
         return self._timestamp_parser(value)
 
     def _do_error_parse(self, response, shape):
-        body = json.loads(response['body'].decode(self.DEFAULT_ENCODING))
-        error = {"Error": {}, "ResponseMetadata": {}}
+        body = self._parse_body_as_json(response['body'])
+        error = {"Error": {"Message": '', "Code": ''}, "ResponseMetadata": {}}
```

Comment:
I can dig into this.  Do you happen to remember any PR/issues?  I see a similar change a while back that added defaults in the parser: https://github.com/boto/botocore/pull/572/files

The only thing that I can remember offhand was that I didn't want *upstream* sources to silently handle these, that is, having the Exception classes handle missing error keys.  Doing that would mean that every layer up would have to gracefully handle the absence of these keys.